src/Agent.Worker/Handlers/AzurePowerShellHandler.cs

```diff
@@ -252,6 +252,12 @@ private void OnDataReceived(object sender, ProcessDataReceivedEventArgs e)
                                                                                       new Tuple<string, string>("Scope", "Global"),
                                                                                   }));
 
+			Trace.Verbose("AzurePowerShellHandler.UpdatePowerShellEnvironment - AddCommand(Set-UserAgent)");
```

Comment:
Another option would be to call Set-UserAgent from within one of the other functions that already gets invoked by the handler.